test/arrays/bradc/noelemtype.good

```diff
@@ -1 +1 @@
-noelemtype.chpl:5: syntax error: near '='
+noelemtype.chpl:5: error: syntax error near '='
```

Comment:
This doesn't seem good to me; I think this indicates that `ERROR` is used instead of `SYNTAX` in the errors. Now that I think about it, the macros for the parser can be adjusted. In my opinion, to the following:

```C++
#define SYNTAX_ERROR_CLASS(NAME, EINFO...) \
  SYNTAX_CLASS(NAME, const Location, ##EINFO)
#define PARSER_ERROR_CLASS(NAME, EINFO...) \
  ERROR_CLASS(NAME, const Location, ##EINFO)
#define PARSER_WARNING_CLASS(NAME, EINFO...) \
  WARNING_CLASS(NAME, const Location, ##EINFO)
```

We need `PARSER_ERROR_CLASS` because semantically, a bison out-of-memory error is not a syntax error, nor is the misc bison error.

Then, the classes can be listed as follows:

```C++
SYNTAX_ERROR_CLASS(Parsing, std::string)
SYNTAX_ERROR_CLASS(CannotAttachPragmas)
SYNTAX_ERROR_CLASS(InvalidIndexExpr)
SYNTAX_ERROR_CLASS(RecordInheritanceNotSupported)
SYNTAX_ERROR_CLASS(InvalidNumericLiteral, std::string, std::string)
SYNTAX_ERROR_CLASS(MultipleExternalRenaming)
PARSER_WARNING_CLASS(PreIncDecOp, std::string, std::string)
SYNTAX_ERROR_CLASS(NewWithoutArgs)
SYNTAX_ERROR_CLASS(UseImportNeedsModule, bool)
SYNTAX_ERROR_CLASS(ExceptOnlyInvalidExpr,
                   uast::VisibilityClause::LimitationKind)
SYNTAX_ERROR_CLASS(LabelIneligibleStmt, const uast::AstNode*)
// Bison* errors are reported by the Bison parser to yyerror
PARSER_ERROR_CLASS(BisonMemoryExhausted)
PARSER_ERROR_CLASS(BisonUnknownError, std::string, std::string)
PARSER_ERROR_CLASS(BisonSyntaxError, std::string)

/* lexer-specific parser errors */
SYNTAX_ERROR_CLASS(StringLiteralEOF)
SYNTAX_ERROR_CLASS(ExternUnclosedPair, std::string)
SYNTAX_ERROR_CLASS(CommentEOF, Location, Location)
```

Reply:
Good catch, that skipped my mind. Should all be switched over now